src/shapes.ts

```diff
@@ -1,33 +1,53 @@
 import { Scene } from "hovl/scene";
 import { Vector } from "hovl/vector";
 
+interface Trs {
+  translation: Vector;
+  rotation: number;
+  scale: Vector | number;
+}
+
 export abstract class Shape {
-  public pos: Vector;
   public color: string;
 
-  constructor(x: number, y: number, color: string) {
-    this.pos = new Vector(x, y);
+  public translation: Vector;
+  public rotation: number;
+  public scaling: Vector;
+
+  constructor(color: string, trs: Partial<Trs>) {
     this.color = color;
+    this.translation = trs.translation || new Vector(0, 0);
+    this.rotation = trs.rotation || 0;
+    this.scaling = typeof trs.scale === "number"
+      ? new Vector(trs.scale, trs.scale)
+      : trs.scale || new Vector(1, 1);
   }
 
-  public abstract draw(scene: Scene): void;
+  public draw(scene: Scene): void {
+    scene.context.save();
+    scene.context.translate(this.translation.x, this.translation.y);
+    scene.context.rotate(this.rotation);
+    scene.context.scale(this.scaling.x, this.scaling.y);
+    this.drawInternal(scene);
+    scene.context.restore();
+  }
+
+  protected abstract drawInternal(scene: Scene): void;
```

Comment:
I kind of dislike the name of this, but to be honest, I can't think of anything better right now.